src/module/data/helpers.mjs

```diff
@@ -69,3 +71,35 @@ export class SizeModel extends foundry.abstract.DataModel {
     return this.value + letter;
   }
 }
+
+/**
+ * Special case StringField which represents a formula.
+ */
+export class FormulaField extends foundry.data.fields.StringField {
+
+  /** @override */
+  static get _defaults() {
+    return foundry.utils.mergeObject(super._defaults, {
+      required: true,
+      deterministic: false
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  _validateType(value) {
+    DSRoll.validate(value);
+    if (this.options.deterministic) {
+      const roll = new Roll(value);
+      if (!roll.isDeterministic) throw new Error("must not contain dice terms");
+    }
+    super._validateType(value);
+  }
+
+  /** @override */
+  _applyChangeAdd(value, delta, model, change) {
+    if (value) return value.concat(" + ", delta);
```

Comment:
We should remove and leading sign (and if the leading sign is "-" use that instead of "+").